src/main/java/com/mraof/minestuck/tileentity/machine/TotemLatheTileEntity.java

```diff
@@ -119,49 +110,30 @@ public boolean setDowel(ItemStack stack)
 		Direction facing = getFacing();
 		BlockPos pos = MSBlocks.TOTEM_LATHE.getDowelPos(getBlockPos(), getBlockState());
 		BlockState state = level.getBlockState(pos);
-		if(stack.isEmpty())
-		{
-			if(isValidDowelRod(state, facing))
-				level.removeBlock(pos, false);
-			return true;
-		} else if(stack.getItem() == MSBlocks.CRUXITE_DOWEL.asItem())
+		BlockState newState = MSBlocks.TOTEM_LATHE.DOWEL_ROD.get().defaultBlockState().setValue(TotemLatheBlock.FACING, facing).setValue(TotemLatheBlock.DowelRod.DOWEL, EnumDowelType.getForDowel(stack));
+		if(isValidDowelRod(state, facing))
 		{
-			BlockState newState = MSBlocks.TOTEM_LATHE.DOWEL_ROD.get().defaultBlockState().setValue(TotemLatheBlock.FACING, facing).setValue(TotemLatheBlock.DowelRod.DOWEL, EnumDowelType.getForDowel(stack));
-			if(isValidDowelRod(state, facing))
-			{
-				TileEntity te = level.getBlockEntity(pos);
-				if(!(te instanceof ItemStackTileEntity))
-				{
-					te = new ItemStackTileEntity();
-					level.setBlockEntity(pos, te);
-				}
-				ItemStackTileEntity teItem = (ItemStackTileEntity) te;
-				teItem.setStack(stack);
-				if(!state.equals(newState))
-					level.setBlockAndUpdate(pos, newState);
-				else level.sendBlockUpdated(pos, state, state, 2);
-				return true;
-			} else if(state.isAir(level, pos))
+			TileEntity te = level.getBlockEntity(pos);
+			if(!(te instanceof ItemStackTileEntity))
 			{
-				level.setBlockAndUpdate(pos, newState);
-				TileEntity te = level.getBlockEntity(pos);
-				if(!(te instanceof ItemStackTileEntity))
-				{
-					te = new ItemStackTileEntity();
-					level.setBlockEntity(pos, te);
-				}
-				ItemStackTileEntity teItem = (ItemStackTileEntity) te;
-				teItem.setStack(stack);
-				
-				return true;
+				te = new ItemStackTileEntity();
+				level.setBlockEntity(pos, te);
 			}
+			ItemStackTileEntity teItem = (ItemStackTileEntity) te;
+			teItem.setStack(stack);
+			if(!state.equals(newState))
+				level.setBlockAndUpdate(pos, newState);
+			else level.sendBlockUpdated(pos, state, state, 2);
+			isProcessing = false;
+			level.sendBlockUpdated(getBlockPos(), getBlockState(), getBlockState(), Constants.BlockFlags.BLOCK_UPDATE);
+			return true;
 		}
 		return false;
 	}
 	
 	public ItemStack getDowel()
 	{
-		BlockPos pos = getBlockPos().above().relative(getFacing().getCounterClockWise(), 2);
+		BlockPos pos = getBlockPos().above().relative(getFacing().getCounterClockWise(), 1);
```

Comment:
Should probably be done with `MSBlocks.TOTEM_LATHE.getDowelPos(getBlockPos(), getBlockState())` instead